what4/src/What4/Solver/CVC5.hs

```diff
@@ -0,0 +1,236 @@
+------------------------------------------------------------------------
+-- |
+-- Module      : What4.Solver.CVC5
+-- Description : Solver adapter code for CVC5
+-- Copyright   : (c) Galois, Inc 2015-2020
+-- License     : BSD3
+-- Maintainer  : Rob Dockins <rdockins@galois.com>
+-- Stability   : provisional
+--
+-- CVC5-specific tweaks to the basic SMTLib2 solver interface.
+------------------------------------------------------------------------
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+
+module What4.Solver.CVC5
+  ( CVC5(..)
+  , cvc5Features
+  , cvc5Adapter
+  , cvc5Path
+  , cvc5Timeout
+  , cvc5Options
+  , runCVC5InOverride
+  , withCVC5
+  , writeCVC5SMT2File
+  , writeMultiAsmpCVC5SMT2File
+  ) where
+
+import           Control.Monad (forM_, when)
+import           Data.Bits
+import           Data.String
+import           System.IO
+import qualified System.IO.Streams as Streams
+
+import           What4.BaseTypes
+import           What4.Concrete
+import           What4.Config
+import           What4.Expr.Builder
+import           What4.Expr.GroundEval
+import           What4.Interface
+import           What4.ProblemFeatures
+import           What4.Protocol.Online
+import qualified What4.Protocol.SMTLib2 as SMT2
+import           What4.Protocol.SMTLib2.Response ( strictSMTParseOpt )
+import qualified What4.Protocol.SMTLib2.Response as RSP
+import qualified What4.Protocol.SMTLib2.Syntax as Syntax
+import           What4.Protocol.SMTWriter
+import           What4.SatResult
+import           What4.Solver.Adapter
+import           What4.Utils.Process
+
+
+intWithRangeOpt :: ConfigOption BaseIntegerType -> Integer -> Integer -> ConfigDesc
+intWithRangeOpt nm lo hi = mkOpt nm sty Nothing Nothing
+  where sty = integerWithRangeOptSty (Inclusive lo) (Inclusive hi)
+
+data CVC5 = CVC5 deriving Show
+
+-- | Path to cvc5
+cvc5Path :: ConfigOption (BaseStringType Unicode)
+cvc5Path = configOption knownRepr "solver.cvc5.path"
+
+cvc5RandomSeed :: ConfigOption BaseIntegerType
+cvc5RandomSeed = configOption knownRepr "solver.cvc5.random-seed"
+
+-- | Per-check timeout, in milliseconds (zero is none)
+cvc5Timeout :: ConfigOption BaseIntegerType
+cvc5Timeout = configOption knownRepr "solver.cvc5.timeout"
+
+-- | Control strict parsing for Boolector solver responses (defaults
```

Comment:
There's a typo here:

```suggestion
-- | Control strict parsing for CVC5 solver responses (defaults
```

In fact, the same typo is also present in `What4.Solver.CVC4`. Would you mind updating that file as well?